src/core/tritonserver.cc

```diff
@@ -1093,69 +1092,66 @@ TRITONSERVER_ServerModelMetadata(
       lserver->ModelReadyVersions(model_name, &ready_versions));
 
   rapidjson::Document metadata;
+  auto& allocator = metadata.GetAllocator();
   metadata.SetObject();
   // Just store string reference in JSON object since it will be serialized to
   // another buffer.
-  metadata.AddMember(
-      "name", rapidjson::StringRef(model_name), metadata.GetAllocator());
+  metadata.AddMember("name", rapidjson::StringRef(model_name), allocator);
 
   rapidjson::Value versions(rapidjson::kArrayType);
   for (const auto v : ready_versions) {
```

Comment:
Don't you need to check and only return one version if `model_version` is specified?

Reply:
Yes. I did not touch that logic but I can fix it in this PR itself if you think thats a good idea,